common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -4,50 +4,267 @@
+               "This key is granted for exclusive use by this course for the specified duration. "
+               "Please do not share the API key with other courses and notify MathWorks immediately "
+               "if you believe the key is exposed or compromised. To obtain a key for your course, "
+               "or to report an issue, please contact moocsupport@mathworks.com"),
+        scope=Scope.settings
+    )
+
+    INDEX_CONTENT_TYPE = 'CAPA'
+
+    resources_dir = None
+
+    has_score = True
+    show_in_read_only_mode = True
+    template_dir_name = 'problem'
+    mako_template = "widgets/problem-edit.html"
+    has_author_view = True
+
+    # The capa format specifies that what we call max_attempts in the code
+    # is the attribute `attempts`. This will do that conversion
+    metadata_translations = dict(XmlParserMixin.metadata_translations)
+    metadata_translations['attempts'] = 'max_attempts'
+
     icon_class = 'problem'
 
-    js = {
+    uses_xmodule_styles_setup = True
+
+    preview_view_js = {
         'js': [
             resource_string(__name__, 'js/src/javascript_loader.js'),
             resource_string(__name__, 'js/src/capa/display.js'),
             resource_string(__name__, 'js/src/collapsible.js'),
             resource_string(__name__, 'js/src/capa/imageinput.js'),
             resource_string(__name__, 'js/src/capa/schematic.js'),
+        ],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js')
+    }
+
+    preview_view_css = {
+        'scss': [
+            resource_string(__name__, 'css/capa/display.scss'),
+        ],
+    }
+
+    studio_view_js = {
+        'js': [
+            resource_string(__name__, 'js/src/problem/edit.js'),
+        ],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js'),
+    }
+
+    studio_view_css = {
+        'scss': [
+            resource_string(__name__, 'css/editor/edit.scss'),
+            resource_string(__name__, 'css/problem/edit.scss'),
         ]
     }
 
-    js_module_name = "Problem"
-    css = {'scss': [resource_string(__name__, 'css/capa/display.scss')]}
+    def student_view(self, _context):
+        """
+        Return the student view.
+        """
+        fragment = Fragment(self.get_html())
+        add_webpack_to_fragment(fragment, 'ProblemBlockPreview')
+        shim_xmodule_js(fragment, 'Problem')
+        return fragment
 
     def author_view(self, context):
         """
         Renders the Studio preview view.
         """
         return self.student_view(context)
 
+    def studio_view(self, _context):
+        """
+        Return the studio view.
+        """
+        fragment = Fragment(
+            self.system.render_template(self.mako_template, self.get_context())
+        )
+        add_webpack_to_fragment(fragment, 'ProblemBlockStudio')
+        shim_xmodule_js(fragment, 'MarkdownEditingDescriptor')
+        return fragment
```

Comment:
This was initially implemented in `MakoTemplateBlockBase` but needed to call `add_webpack_to_fragment()` and `shim_xmodule_js()` with custom parameters so copied over.

Reply:
Both `add_webpack_to_fragment()` and `shim_xmodule_js` are called in `openedx/core/lib/xblock_utils/__init__.py` but that works because the Module and Descriptor are two different classes. In the interests to making this more XBlock like, have moved calling these to in `student_view()` and `studio_view()`.